sql/analyzer/validation_rules.go

```diff
@@ -246,10 +246,29 @@ func validateGroupBy(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (s
 			groupBys = append(groupBys, expr.String())
 		}
 
+		var curAliases = make(map[string]sql.Expression)
+		if p, ok := n.Child.(*plan.Project); ok {
```

Comment:
there are a variety of node types that could be a `*plan.GroupBy` child. An order by might insert a sort node, for example:
```
tmp3> explain select * from b group by x order by x;
+---------------------------------------+
| plan                                  |
+---------------------------------------+
| GroupBy                               |
|  ├─ SelectedExprs(b.x)                |
|  ├─ Grouping(b.x)                     |
|  └─ Sort(b.x ASC)                     |
|      └─ Projected table access on [x] |
|          └─ Exchange(parallelism=12)  |
|              └─ Table(b)              |
+---------------------------------------+
```

Reply:
In the context of validating group by functional dependencies, this means 1) a node TraverseUp might be necessary to resolve all of the child nodes, and 2) an expression Traverse up might be necessary to find the expression dependencies for each child. This [copy function](https://github.com/dolthub/go-mysql-server/blob/30a0c6358b4500db684b0d3fb478521c9a057d66/sql/analyzer/analyzer.go#L507) (not merged yet) shows one way to do this sort of traversal.

Reply:
Any way you look at this, the above plan is a bug. `ORDER BY` always applies to a the final result set, it cannot be the child of something like a grouping operation. Can you file a bug for that @max-hoffman ?

Reply:
If we had a grouping operator that took advantage of sorted input it would make sense, but otherwise yeah.